### PR TITLE
Add new warning ``deprecated-class`` #4388

### LINKED ISSUE #4388: Add _collections_abc methods to DEPRECATED_METHODS
### Is your feature request related to a problem? Please describe

According to Python documentation of module `collections`[1], methods in `_collections_abc` will be moved from `collections` to `collections.abc`. And now Pylint hasn't checked it.

### Describe the solution you'd like

I suggest to add methods in `_collections_abc.__all__` to `DEPRECATED_METHODS`[3] of Pylint.

### Additional context

Test code is 
```
from collections import Iterable


print(isinstance(None, Iterable))
```

pylint 2.7.4
astroid 2.5.3
Python 3.7.2 

[1]https://docs.python.org/3/library/collections.html
[2]https://github.com/python/cpython/blob/3.9/Lib/_collections_abc.py#L18
[3]https://github.com/PyCQA/pylint/blob/af52033971eccecea47597ebbfaeac15773b3e1b/pylint/checkers/stdlib.py#L85

### PR DESCRIPTION
**Pull request by jiajunsu**
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
According to Python documentation of module `collections`[1], classes in `_collections_abc`[2] will be moved from `collections` to `collections.abc`. And now Pylint hasn't checked it.

This PR add ``deprecated-class`` to check deprecated class that be imported or called, and could check the deprecation above.

**Unsupport scenario**
I haven't found a proper method that could detect the code below. Maybe it could be implemented in future.

```
import collections


isinstance(None, collections.Iterable)
```

[1]https://docs.python.org/3/library/collections.html
[2]https://github.com/python/cpython/blob/3.9/Lib/_collections_abc.py#L18

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Related Issue

Closes #4388

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -17,6 +17,11 @@ Release date: 2021-04-26
 
 * Pylint's tags are now the standard form ``vX.Y.Z`` and not ``pylint-X.Y.Z`` anymore.
 
+* New warning message ``deprecated-class``. This message is emitted if import or call deprecated class of the
+__standard library (like ``collections.Iterable`` that will be removed in Python 3.10).
```
Comment (Pierre-Sassoulas) [inline ChangeLog:21]: I thought collection.abc was going to stop working in 3.9 ?

File: ChangeLog [hunk 2]
```diff
@@ -17,6 +17,11 @@ Release date: 2021-04-26
...
+
+__Closes #4388
```
Comment (Pierre-Sassoulas) [inline ChangeLog:23]:
```suggestion
  Closes #4388
```
There is the same underscore before __standard library

File: tests/functional/d/deprecated/deprecated_class_py33.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# pylint: disable=unused-import,import-error,no-name-in-module
+
+from collections import Iterable  # [deprecated-class]
+
+import collections.Set  # [deprecated-class]
+
+import collections
+
+
+_ = collections.Awaitable()  # [deprecated-class]
```
Comment (Pierre-Sassoulas) [inline tests/functional/d/deprecated/deprecated_class_py33.py:11]: Good functional tests !

File: pylint/checkers/deprecated.py [hunk 4]
```diff
@@ -154,3 +181,38 @@ def check_deprecated_method(self, node, inferred):
                 self.add_message(
                     "deprecated-argument", node=node, args=(arg_name, func_name)
                 )
+
+    def check_deprecated_class(self, node, mod_path=""):
+        """Checks if the class is deprecated"""
+
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/deprecated.py:187]: I think we can reduce duplication by getting mod_name or mod_path and class_name before and then doing only one 
```python
if class_name in self.deprecated_classes(mod_name_or_path):
    self.add_message(
        "deprecated-class", node=node, args=(class_name, mod_name_or_path)
    )
 ```
 Like in https://github.com/PyCQA/pylint/pull/4419/files#diff-7d982c7ee8d920542a08ef8e7a73fddf8929599d2c13507a7b3b6bf2a4371b6fR157

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
👍 I made some minor comments. Thank you for your contribution !

File: ChangeLog (hunk 1 above)
Comment (jiajunsu) [reply ChangeLog:21]: I checked the cpython code in 3.9[1], ``from collections import Iterable`` still works. And in master(3.10), the function ``__getattr__`` in ``collections.__init__.py`` has been removed.

https://github.com/python/cpython/blob/3.9/Lib/collections/__init__.py#L55

**Comment by jiajunsu** [comment]
> 👍 I made some minor comments. Thank you for your contribution !

Thanks for your suggestions, I have refactored the function ``check_deprecated_class`` :)

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you, great job with the rework !

**Comment by jiajunsu** [comment]
pre-commit.ci run

**Comment by Pierre-Sassoulas** [comment]
I'm going to fix the pre-commit ci job later, don't worry about it :)

**Comment by jiajunsu** [comment]
> I'm going to fix the pre-commit ci job later, don't worry about it :)

Thanks, I was just debugging :)

**Comment by Pierre-Sassoulas** [comment]
Apparentely you could add:
```yaml
ci:
    skip: [pylint]
```
On top of the pre-commit config file.

**Comment by Pierre-Sassoulas** [comment]
Right now there is redundancy with github action about this so it's not a problem.

**Comment by jiajunsu** [comment]
> Apparentely you could add:
> 
> ```yaml
> ci:
>     skip: [pylint]
> ```
> 
> On top of the pre-commit config file.

I've added that.

**Comment by Pierre-Sassoulas** [comment]
Thank you for also fixing the pipeline for us on top of the MR ! And congratulation on becoming a pylint contributor !

**Pull request merged**